plugins/kubernetes/app/models/kubernetes/release_doc.rb

```diff
@@ -36,36 +33,52 @@ def service
       kubernetes_role.service_for(deploy_group) if has_service?
     end
 
-    def pretty_rc_doc(format: :json)
-      case format
-        when :json
-          JSON.pretty_generate(deployment_hash)
-        when :yaml, :yml
-          deployment_hash.to_yaml
-        else
-          deployment_hash.to_s
-      end
-    end
-
     def build
       kubernetes_release.try(:build)
     end
 
+    def release_doc_metadata
+      kubernetes_release.release_metadata.merge(role_metadata).merge(deploy_group_metadata)
+    end
+
+    def role_metadata
+      { role_id: kubernetes_role.id.to_s, role_name: kubernetes_role.name }
+    end
+
+    def deploy_group_metadata
+      { deploy_group_id: deploy_group.id.to_s, deploy_group_namespace: deploy_group.kubernetes_namespace }
+    end
+
     def nested_error_messages
       errors.full_messages
     end
 
-    def update_replica_count(new_count)
-      self.replicas_live = new_count
+    def update_release
+      kubernetes_release.update_status(self)
+    end
 
-      if replicas_live >= replica_target
-        self.status ='live'
-      elsif replicas_live > 0 && !failed?
-        self.status ='spinning_up'
+    def update_status(live_pods)
+      case
+      when live_pods == replica_target then self.status = :live
+      when live_pods.zero? then self.status = :dead
+      when live_pods > replicas_live then self.status = :spinning_up
+      when live_pods < replicas_live then self.status = :spinning_down
```

Comment:
Wondering if this might lead to confusion if user has just deployed this release, and pods are crashing and getting restarted, the status might get set to 'spinning down'?